open_learning_ai_tutor/prompts.py

```diff
@@ -0,0 +1,66 @@
+
+
+intent_mapping = {
+    Intent.P_LIMITS: "Make the student identify the limits of their reasoning or answer by asking them questions.\n",
+    Intent.P_GENERALIZATION: "Ask the student to generalize their answer.\n",
+    Intent.P_HYPOTHESIS: "Ask the student to start by providing a guess or explain their intuition of the problem.\n",
+    Intent.P_ARTICULATION: "Ask the student to write their intuition mathematically or detail their answer.\n",
+    Intent.P_REFLECTION: "Step back and reflect on the solution. Ask to recapitulate and *briefly* underline more general implications and connections.\n",
+    Intent.P_CONNECTION: "Underline the implication of the answer in the context of the problem.\n",
+    Intent.S_SELFCORRECTION: "If there is a mistake in the student's answer, tell the student there is a mistake in an encouraging way and make them identify it *by themself*.\n",
+    Intent.S_CORRECTION: "Correct the student's mistake if there is one, by stating or hinting them what is wrong.\nConsider the student's mistake, if there is one.\n",
+    Intent.S_STRATEGY: "Acknowledge the progress. Encourage and make the student find on their own what is the next step to solve the problem, for example by asking a question. You can also move on to the next part\n",
+    Intent.S_HINT: "Give a hint to the student to help them find the next step. Do *not* provide the answer.\n",
+    Intent.S_SIMPLIFY: "Consider first a simpler version of the problem.\n",
+    Intent.S_STATE: "State the theorem, definition or programming command the student is asking about. You can use the whiteboard tool to explain. Keep the original exercise in mind. DO NOT REVEAL ANY PART OF THE EXERCISE'S SOLUTION: use other examples.\n",
+    Intent.S_CALCULATION: "Correct and perform the numerical computation for the student.\nConsider the student's mistake, if there is one.\n",
+    Intent.A_CHALLENGE: "Maintain a sense of challenge.\n",
+    Intent.A_CONFIDENCE: "Bolster the student's confidence.\n",
+    Intent.A_CONTROL: "Promote a sense of control.\n",
+    Intent.A_CURIOSITY: "Evoke curiosity.\n",
+    Intent.G_GREETINGS: "Say goodbye and end the conversation\n",
+    Intent.G_OTHER: "",
+}
+
+
+def get_intent_prompt(intents):
+    intent_prompt = ""
+
+    if Intent.G_REFUSE in intents:
+        intent_prompt = "The student is asking something irrelevant to the problem. Explain politely that you can't help them on topics other than the problem. DO NOT ANSWER THEIR REQUEST\n"
+    else:
+        for intent in intents:
+            intent_prompt += intent_mapping.get(intent, "")
+
+    intent_prompt += " Your student can only see the text you send them using your `text_student` tool, the rest of your thinking is hidden to them."
```

Comment:
This shouldn't be needed after i refactor the graph in Tutor.py but leaving for now